### PR TITLE
create dropdown for course types

### LINKED ISSUE #323: Create dropdown for accesskey, whitelist, free course
# User Story

### As a: 
Teacher

### I want:
a dropdown menu for the type of my course

### so that:
I can assign the correct access rights to the course

### PR DESCRIPTION
**Pull request by jsinisa**
## Description:
With the dropdown you can select the course type and become the right possibility for entering your students
closes #323 

## Improvements
Better select possibility for the course type and the right method to entering the course for the selected course type

### CODE REVIEW

**Comment by jsinisa** [comment]
@grosspersky do want some changes ?

File: app/webFrontend/src/app/course/course-edit/course-edit.component.html [hunk 1]
```diff
@@ -14,18 +14,33 @@ <h1>Edit course</h1>
...
 
-        <mat-form-field>
+        <mat-form-field class="full-width">
+          <mat-select name="courseTypes" placeholder="Please select your course type" id="courseTypes" [(ngModel)]="enrollType" (change)="selectType($event)">
+            <mat-option *ngFor="let enrollType of enrollTypes" [value]="enrollType">{{enrollType}}</mat-option>
+          </mat-select><br>
+        </mat-form-field>
+
+
+        <p *ngIf="enrollType === 'free'">The course is free and accessible for everyone</p>
```
Comment (lukaskorte) [inline app/webFrontend/src/app/course/course-edit/course-edit.component.html]: Do not use string literals in template, the constants should also be used here

File: app/webFrontend/src/app/course/course-edit/course-edit.component.html [hunk 2]
```diff
@@ -14,18 +14,33 @@ <h1>Edit course</h1>
...
+
+
+        <mat-form-field class="full-width" *ngIf="enrollType === 'accesskey'">
+          <p>Enter a accesskey for the students</p>
```
Comment (lukaskorte) [inline app/webFrontend/src/app/course/course-edit/course-edit.component.html]: This should be a hint label: https://material.angular.io/components/form-field/overview#hint-labels

File: app/webFrontend/src/app/course/course-edit/course-edit.component.html [hunk 3]
```diff
@@ -14,18 +14,33 @@ <h1>Edit course</h1>
...
+          <p>
           <input async matInput id="accessKey" placeholder="Access Key" [(ngModel)]="accessKey"
                  [ngModelOptions]="{standalone: true}">
+          </p>
         </mat-form-field>
 
+        <!--
```
Comment (lukaskorte) [inline app/webFrontend/src/app/course/course-edit/course-edit.component.html]: Do not commit commented code

File: app/webFrontend/src/app/course/course-edit/course-edit.component.html [hunk 4]
```diff
@@ -14,18 +14,33 @@ <h1>Edit course</h1>
...
                          [checked]="mode">
           <span *ngIf="enrollType === 'free' && !accessKey">Everyone can enroll</span>
           <span
             *ngIf="enrollType === 'free' && accessKey">Everyone can enroll using the access key</span>
           <span *ngIf="enrollType === 'whitelist'">Only whitelisted members can enroll</span>
         </mat-slide-toggle>
+        -->
 
         <div class="csv-upload" *ngIf="enrollType === 'whitelist'">
           <p>Upload your OBS csv file with your course members for whitelisting students.</p>
```
Comment (lukaskorte) [inline app/webFrontend/src/app/course/course-edit/course-edit.component.html]: The dropdown should make this code below unnecessary:
```
          <mat-slide-toggle (change)="onChangeActive($event)"
                            [checked]="active">
             <span>Course can be seen by students</span>
           </mat-slide-toggle>
```

File: app/webFrontend/src/app/course/course-edit/course-edit.component.ts [hunk 5]
```diff
@@ -117,6 +125,19 @@ export class CourseEditComponent implements OnInit {
     }
   }
 
+  selectType(event) {
+    console.log(event);
```
Comment (lukaskorte) [inline app/webFrontend/src/app/course/course-edit/course-edit.component.ts]: Do not commit logs

File: app/webFrontend/src/app/course/course-edit/course-edit.component.ts [hunk 6]
```diff
@@ -117,6 +125,19 @@ export class CourseEditComponent implements OnInit {
...
+    if (event.value === ENROLL_TYPE_WHITELIST) {
+      this.enrollType = event.value;
```
Comment (lukaskorte) [inline app/webFrontend/src/app/course/course-edit/course-edit.component.ts]: Do not duplicate this code for every case

File: app/webFrontend/src/app/course/course-edit/course-edit.component.ts [hunk 7]
```diff
@@ -117,6 +125,19 @@ export class CourseEditComponent implements OnInit {
...
+    }
+    if (event.value === ENROLL_TYPE_FREE) {
+      this.enrollType = event.value;
+    }
+    if (event.value === ENROLL_TYPE_ACCESSKEY) {
+      this.enrollType = event.value;
```
Comment (lukaskorte) [inline app/webFrontend/src/app/course/course-edit/course-edit.component.ts]: ⚠⚠⚠ This type is not handled by the API, we cannot merge this before this is the case ⚠⚠⚠

**Pull request merged**